File: iosApp/iosApp/Presentation/Community/Detail/PostDetailUiState.swift
//
//  PostDetailUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 4/28/26.
//

import Foundation
import Shared

struct PostDetailUiState {
    var post: CommunityPost? = nil
    var isLoading: Bool = true
    var isLiked: Bool = false
    var isOwner: Bool = false
    var currentUserId: String? = nil
    var isMenuVisible: Bool = false
    var isDeleteConfirmVisible: Bool = false
    var isReportSheetVisible: Bool = false
    var reportingCommentId: String? = nil
    var selectedReportType: String? = nil
    var isSubmittingReport: Bool = false
    var isDeleting: Bool = false
    var comments: [Comment] = []
    var isLoadingComments: Bool = false
    var hasMoreComments: Bool = false
    var isLoadingMoreComments: Bool = false
    var oldestCommentCursor: String? = nil
    var editingCommentId: String? = nil
    var editCommentText: String = ""
    var isUpdatingComment: Bool = false
    var commentText: String = ""
    var isSendingComment: Bool = false
    var errorMessage: String? = nil

    var likeCount: Int { Int(post?.likeCount ?? 0) }
    var commentCount: Int { Int(post?.commentCount ?? 0) }
    var canSendComment: Bool { !commentText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty && !isSendingComment }
    var canUpdateComment: Bool { !editCommentText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty && !isUpdatingComment }
}

File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/community/detail/PostDetailUiState.kt
package com.hhp227.concafe.presentation.main.community.detail

import com.hhp227.concafe.domain.model.Comment
import com.hhp227.concafe.domain.model.CommunityPost

data class PostDetailUiState(
    val post: CommunityPost? = null,
    val isLoading: Boolean = false,
    val isLiked: Boolean = false,
    val isOwner: Boolean = false,
    val currentUserId: String? = null,
    val isMenuVisible: Boolean = false,
    val isDeleteConfirmVisible: Boolean = false,
    val isReportSheetVisible: Boolean = false,
    val reportingCommentId: String? = null,
    val selectedReportType: String? = null,
    val isSubmittingReport: Boolean = false,
    val isBlockingUser: Boolean = false,
    val blockedUserIds: Set<String> = emptySet(),
    val isDeleting: Boolean = false,
    val comments: List<Comment> = emptyList(),
    val isLoadingComments: Boolean = false,
    val hasMoreComments: Boolean = false,
    val isLoadingMoreComments: Boolean = false,
    val oldestCommentCursor: String? = null,
    val editingCommentId: String? = null,
    val editCommentText: String = "",
    val isUpdatingComment: Boolean = false,
    val commentText: String = "",
    val isSendingComment: Boolean = false,
    val errorMessage: String? = null
) {
    val likeCount: Int get() = post?.likeCount ?: 0
    val commentCount: Int get() = post?.commentCount ?: 0
    val canSendComment: Boolean get() = commentText.isNotBlank() && !isSendingComment
    val canUpdateComment: Boolean get() = editCommentText.isNotBlank() && !isUpdatingComment
}
